### ISSUE TITLE
Feature request - Configure show/hide at the picker level

### ISSUE THREAD

**Issue by p-m-j** [AndyButland/UmbracoPersonalisationGroupsCore#15]
Hey Andy, hope you're well! 

First of all thanks for the awesome plugin!

It might be inexperience with the plugin but I feel like there is a focus on showing additional content vs hiding existing content.

I can see that most criteria have options such as querystring value contains vs does not contain but what would be really handy for me is the ability to define a group as is but via the picker choose the outcome for that group.

I don't know if the image is useful but this is what i'd like to achieve, using the same group for both blocks but one block should be shown and the other hidden.

![image](https://github.com/AndyButland/UmbracoPersonalisationGroupsCore/assets/2056399/ce654c48-d5ff-430d-aa5f-4196bcfc4369)

(I would be open to contributing if you think there is value here)

**Comment by AndyButland** [comment]
Hi Paul - good to hear from you, hope you are doing well too.  "Awesome" is pushing it a bit I think, but seems like it meets a need in a small way.

I see your point, but one issue getting there is that currently the picker you associated with your blocks is just a standard content picker - it's not something custom that comes with the package.  So we'd need to introduce that to allow you to choose "show/hide" on your block itself as well as picking the group.

Wondering if you can meet the need just by creating two groups, one the opposite of the other?  So you could pick "matches X" for one block and "does not match X" for another?

**Comment by p-m-j** [comment]
Right now our needs are 100% met by the out of the box functionality.

**Requirement:** Hide specified navigation items when a cookie is present with a given value.
**Solution:** cookie criteria with "Does not match regular expression".

But it feels backwards, I want to configure "if you have this cookie value you belong to this group" then later "if you are in this group this behaviour is applied"

----

This is 100% an enhancement idea for the future.

Adding a new property editor sounds like it would be the right way to go + some changes in dotnet land such that the result of `ShowToVisitor` is negated based on the configuration.

**Comment by p-m-j** [comment]
@AndyButland - on a related note I wonder if behaviour for DoesNotMatchRegex is "correct".

```csharp
case CookieSettingMatch.DoesNotMatchRegex:
    return cookieExists && !MatchesRegex(cookieValue, cookieSetting.Value);
```

If the cookie doesn't exist, it cannot match the regex so I'd hope for a return value of true.

**Comment by AndyButland** [comment]
I agree @p-m-j - I think this classes as a bug, so I've put out a patch to resolve this.  See [3.2.10](https://www.nuget.org/packages/UmbracoPersonalisationGroups/3.2.10).

**Comment by p-m-j** [comment]
> I agree @p-m-j - I think this classes as a bug, so I've put out a patch to resolve this. See [3.2.10](https://www.nuget.org/packages/UmbracoPersonalisationGroups/3.2.10).

Thanks Andy that's great news!
